### PR TITLE
Parameterize some metric tests

### LINKED ISSUE #1469: Parametrize metric tests
## 🚀 Feature

Use `pytest.mark.parametrize` for metric testing

### PR DESCRIPTION
**Pull request by Moh-Yakoub**
Fixes #1469

Description:

Parameterize some metric tests.

I started with couple of test cases that had obvious use of test parameterization. Will refactor more metric tests in later PRs

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/metrics/test_loss.py [hunk 1]
```diff
@@ -135,16 +122,6 @@ def _test_distrib_accumulator_device(device):
         ), f"{type(loss._sum.device)}:{loss._sum.device} vs {type(metric_device)}:{metric_device}"
 
 
-def test_sum_detached():
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_loss.py:138]: Why did you remove this test ?

File: tests/ignite/metrics/test_loss.py [hunk 2]
```diff
@@ -17,22 +17,9 @@ def test_zero_div():
...
-
-    y_pred = torch.tensor([[0.1, 0.3, 0.6], [0.6, 0.2, 0.2], [0.2, 0.7, 0.1]]).log()
-    y = torch.tensor([2, 0, 2]).long()
-    loss.update((y_pred, y))
-    assert_almost_equal(loss.compute(), 1.1253643036)  # average
-
-
-def test_compute_on_criterion():
-    loss = Loss(nn.NLLLoss())
+@pytest.mark.parametrize("loss_function", [nll_loss, nn.NLLLoss()])
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_loss.py:20]: This is a good usage 👍

**Review by vfdev-5 — COMMENTED** [review]
Thanks for experimenting with parametrize @Moh-Yakoub 
IMO, this is still something to think about on how to use it properly. I left my comments on the PR.
